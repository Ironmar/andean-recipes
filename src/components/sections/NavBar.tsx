import React, { useState } from "react";
import tinkin from "../../assets/images/tinkin.png";
import add from "../../assets/images/add_icon.svg";
import FormModal from "../generics/FormModal";
const Navbar: React.FC = () => {
  const [showModal, setShowModal] = useState<boolean>(false);

  return (
    <>
      {showModal && <FormModal setShowModal={setShowModal} />}
      <div className="nav">
        <button
          className="add-button"
          type="button"
          onClick={() => setShowModal(true)}
        >
          <img src={add} alt="add_icon" />
          Add Recipe
        </button>
        <img className="logo" src={tinkin} alt="tinkin_logo" />
      </div>
    </>
  );
};
export default Navbar;
