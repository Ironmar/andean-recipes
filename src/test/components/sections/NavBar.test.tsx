import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";

import "@testing-library/jest-dom";
import NavBar from "../../../components/sections/NavBar";
import tinkin from "../../../assets/images/tinkin.png";
configure({ adapter: new Adapter() });
//@ts-ignore
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Test in <NavBar/>", () => {
  let wrapper: ShallowWrapper<any, Readonly<{}>>;

  beforeEach(() => {
    wrapper = shallow(<NavBar />);
  });
  test("should be return the component correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("shuold be return the correct logo path", () => {
    const image = wrapper.find(".logo");
    expect(image.prop("src")).toEqual(tinkin);
    expect(image.prop("alt")).toEqual("tinkin_logo");
  });

  test("should be return the text of button", () => {
    const text = wrapper.find(".add-button");
    expect(text.text()).toEqual("Add Recipe");
  });
});
