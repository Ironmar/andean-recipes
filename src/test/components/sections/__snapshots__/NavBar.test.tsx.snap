// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <NavBar/> should be return the component correctly 1`] = `
<Fragment>
  <div
    className="nav"
  >
    <button
      className="add-button"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="add_icon"
        src="add_icon.svg"
      />
      Add Recipe
    </button>
    <img
      alt="tinkin_logo"
      className="logo"
      src="tinkin.png"
    />
  </div>
</Fragment>
`;
